vms/platformvm/txs/executor/staker_tx_verification.go

```diff
@@ -589,23 +550,22 @@ func verifyAddPermissionlessDelegatorTx(
 		return nil
 	}
 
-	currentTimestamp := chainState.GetTimestamp()
-	// Ensure the proposed validator starts after the current timestamp
-	startTime := tx.StartTime()
-	if !currentTimestamp.Before(startTime) {
-		return fmt.Errorf(
-			"chain timestamp (%s) not before validator's start time (%s)",
-			currentTimestamp,
-			startTime,
-		)
+	var (
+		currentTimestamp = chainState.GetTimestamp()
+		startTime        = tx.StartTime()
+		duration         = tx.Validator.Duration()
+	)
+
+	// Ensure the proposed validator starts after the current time
+	if err := checkStakerStartTime(currentTimestamp, startTime); err != nil {
+		return err
 	}
 
 	delegatorRules, err := getDelegatorRules(backend, chainState, tx.Subnet)
 	if err != nil {
 		return err
 	}
 
-	duration := tx.Validator.Duration()
```

Comment:
nit: can we restore the declaration here? we typically declare variables close to where they're used

Reply:
it was an attempt at reducing diffs but you're right, bette declare close to use